x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlTasks.java

```diff
@@ -88,6 +89,20 @@ public static String snapshotUpgradeTaskId(String jobId, String snapshotId) {
         return JOB_SNAPSHOT_UPGRADE_TASK_ID_PREFIX + jobId + "-" + snapshotId;
     }
 
+    public static Tuple<String, String> jobAndSnapshotId(String snapshotUpgradeTaskId) {
```

Comment:
I don't think this is necessary as the place that uses it has access to the persistent task params.